crates/pathfinder/src/rpc/serde.rs

```diff
@@ -178,7 +299,41 @@ pub(crate) fn bytes_to_hex_str(bytes: &[u8]) -> String {
         buf[pos..pos + 2].copy_from_slice(&x);
     });
     buf[1] = b'x';
+    &buf[..len]
+}
 
+/// A convenience function which produces a "0x" prefixed hex str slice in a given buffer `buf`
+/// from an array of bytes.
+/// Returns `InvalidBufferLengthError` if `bytes.len() * 2 + 2 > buf.len()`
+pub(crate) fn bytes_as_hex_str<'a>(
+    bytes: &'a [u8],
+    buf: &'a mut [u8],
+) -> Result<Cow<'a, str>, InvalidBufferSizeError> {
+    if bytes.len() * 2 + 2 > buf.len() {
+        return Err(InvalidBufferSizeError {
+            actual: buf.len(),
+            expected: bytes.len() * 2 + 2,
+        });
+    }
+
+    if !bytes.iter().any(|b| *b != 0) {
+        return Ok(Cow::from("0x0"));
+    }
+
+    let (it, start, len) = skip_zeros(bytes);
+    let res = it_to_hex_str(it, start, len, buf);
+    // No heap allocations
+    Ok(String::from_utf8_lossy(res))
+}
+
+/// A convenience function which produces a "0x" prefixed hex string from a [StarkHash].
+pub(crate) fn bytes_to_hex_str(bytes: &[u8]) -> String {
+    if !bytes.iter().any(|b| *b != 0) {
+        return "0x0".to_string();
```

Comment:
Perhaps this fn would be a more natural place to use the `Cow<'static, str>`.

Reply:
This one is supposed to produce an owned result as it's used in situations where the local buffer would go out of scope. So I would stay with `String`.

Reply:
And as you pointed out it's just more intuitive to use `&str` in the other, `as_` flavor, instead of `Cow<'_, str>`.